### PR TITLE
Add French holidays

### PR DESCRIPTION
**Pull request by levrailoup**

### CODE REVIEW

**Comment by wblondel** [comment]
Hi @levrailoup 
According to [this page](https://travail-emploi.gouv.fr/droit-du-travail/les-conges-payes-et-les-conges-pour-projets-pro-et-perso/article/les-jours-feries-et-les-ponts), these holidays also need to be included:

- lundi de Pâques
- Ascension
- lundi de Pentecôte

Also, in the departments of Moselle, Bas-Rhin and Haut-Rhin, the list is slightly different: https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000006902635

In Guadeloupe, Guyane, Martinique, Mayotte, La Réunion, Saint-Barthélemy and Saint-Martin, which are all part of France, there are also additional holidays: https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000035902463

I am not sure if this package handles regional differences.

**Comment by levrailoup** [comment]
Lundi de Pentecôte, Ascension and Lundi de Pâques are already included in the PR : As they are dependent upon Easter date, they can be found in France::variableHolidays().

As for the other dates you mention... I'm not quite sure what is expected from this package : should we have a specific class for each region ? Country codes don't include regions AFAIK, so it might prove difficult. "Vendredi Saint" is even town-specific...

In the meantime, the current PR includes all common holidays for mainland France.

File: src/Countries/France.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            'Armistice 1918' => '11-11',
+            'Noël' => '12-25',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
+            ->setTimezone('Europe/Brussels');
```
Comment (Kocal) [inline src/Countries/France.php:33]: Why not using the Paris timezone?

```suggestion
            ->setTimezone('Europe/Paris');
```

Comment (levrailoup) [reply src/Countries/France.php:33]: You are absolutely right

File: src/Countries/France.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+use Carbon\CarbonImmutable;
+
+class France extends Country
+{
+    public function countryCode(): string
+    {
+        return 'fr';
+    }
+
+    /** @return array<string, CarbonImmutable> */
```
Comment (stof) [inline src/Countries/France.php:14]: this return type is wrong (but anyway, you should not need to specify it here. Inheriting it from the parent is better)

**Comment by Nielsvanpach** [comment]
I've added documentation on how I think we should manage regions or states:
https://github.com/spatie/holidays?tab=readme-ov-file#contributing-a-new-country

**Comment by wblondel** [comment]
> Lundi de Pentecôte, Ascension and Lundi de Pâques are already included in the PR : As they are dependent upon Easter date, they can be found in France::variableHolidays().

My bad. I must have forgotten to open my eyes 😅

**Comment by Nielsvanpach** [comment]
Could you rebase this branch, I think it should allow the tests to run here

**Comment by Nielsvanpach** [comment]
This PR looks good! I'll wait for the department based holidays before merging this.

I've added documentation on how I think we should manage regions, states or departments:
https://github.com/spatie/holidays?tab=readme-ov-file#contributing-a-new-country

**Comment by levrailoup** [comment]
Added support for region-specific holidays. It uses [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2:FR) codes for now, but I am open to suggestions.

**Comment by Nielsvanpach** [comment]
Thanks for your work on this one!

**Pull request merged**